File: Try_GameEngine5/src/com/example/try_gameengine/action/visitor/MovementActionObjectStructure.java
package com.example.try_gameengine.action.visitor;

import com.example.try_gameengine.action.MovementAction;

/**
 * {@code MovementActionObjectStructure} is use for set strucuar
 * @author irons
 *
 */
public class MovementActionObjectStructure {
	private MovementAction root = null;
	
	public void handleRequest(IMovementActionVisitor movementActionVisitor){
		if(root != null){
			root.accept(movementActionVisitor);
		}
	}
	
	public void setRoot(MovementAction movementAction){
		this.root = movementAction;
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/action/visitor/MovementActionObjectStructure.kt
package com.example.try_gameengine.action.visitor

import com.example.try_gameengine.action.MovementAction

/**
 * `MovementActionObjectStructure` is use for set strucuar
 * @author irons
 // */
class MovementActionObjectStructure {
    private var root: MovementAction? = null

    fun handleRequest(movementActionVisitor: IMovementActionVisitor?) {
        if (root != null) {
            root!!.accept(movementActionVisitor!!)
        }
    }

    fun setRoot(movementAction: MovementAction?) {
        this.root = movementAction
    }
}
